### PR TITLE
python312Packages.chart-studio: Remove nose dependency

### PR DESCRIPTION
**Pull request by pyrox0**
## Description of changes
Part of #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (emilazy) [inline pkgs/development/python-modules/chart-studio/default.nix:42]: Do we need both of these?

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (emilazy) [inline pkgs/development/python-modules/chart-studio/default.nix:44]: Nit: a newline here would be nice.

**Review by emilazy — CHANGES_REQUESTED** [review]
This is a weird one: it hasn’t seen a release in years but we just get it from the release tags of the Git repository it’s colocated in. I think we should change `version` to `"1.1.0-unstable-YYYY-MM-DD"` to reflect the actual last PyPI release version, but it’s okay to keep getting it from Git because it contains more fixes. One of those fixes is removing the `six` dependency, so let’s get rid of it here too.

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (pyrox0) [reply pkgs/development/python-modules/chart-studio/default.nix:42]: Nope, will remove pytest

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (pyrox0) [reply pkgs/development/python-modules/chart-studio/default.nix]: Added one

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (emilazy) [inline pkgs/development/python-modules/chart-studio/default.nix:15]:
```suggestion
  version = "1.1.0-unstable-2024-07-23";
```

The `version` here is the version of the `plotly` package that happens to share a repository with this; the latest PyPI release is actually [1.1.0](https://pypi.org/project/chart-studio/). We should use the versioning scheme of the actual package to avoid confusing users here.

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (emilazy) [inline pkgs/development/python-modules/chart-studio/default.nix:25]: Given my requested `version` change, you can just inline the tag version directly here, or `let plotlyVersion = …;`, or even just use `plotly.version` (but I wouldn’t necessarily recommend that; moving `plotly` over to `fetchFromGitHub` and inheriting its entire `src` could work, though).

**Review by emilazy — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/chart-studio/default.nix
Comment (pyrox0) [reply pkgs/development/python-modules/chart-studio/default.nix:25]: Moving plotly over to fetchFromGitHub is a larger task than I want to do in this pr as it needs NPM dependencies. Let's merge this and revisit later if this is really a big issue. I'll update to use the chart-studio version in the public version number but plotly's version number for the git tag.

Comment (emilazy) [reply pkgs/development/python-modules/chart-studio/default.nix]: I agree that it’s probably best not to couple the sources for now. However, in that case we probably shouldn’t inherit from `plotly`’s version, because the FOD hash won’t be updated when `plotly` does, causing confusing and nondeterministic results depending on what previous versions you’ve built. I think we should just hardcode the tag for now.

Comment (pyrox0) [reply pkgs/development/python-modules/chart-studio/default.nix]: Tag hardcoded, we can revisit this when/if plotly gets moved to fetchFromGitHub

**Comment by emilazy** [comment]
There’s a conflict because @Aleksanaa beat you to the `maintainers` clean‐up 🫠

**Review by emilazy — CHANGES_REQUESTED** [review]
This needs a rebase because of https://github.com/NixOS/nixpkgs/pull/332440. Sorry for the slow review.

(Kind of funny that @r-ryantm will make no‐op “updates” like that to a directory that hasn’t been touched in 2 years 🫠)

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 330248` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.chart-studio</li>
    <li>python311Packages.chart-studio.dist</li>
    <li>python312Packages.chart-studio</li>
    <li>python312Packages.chart-studio.dist</li>
  </ul>
</details>

**Pull request merged**
